### PR TITLE
alpaka CI: set CMAKE_CXX_COMPILER instead using update-alternatives

### PR DESCRIPTION
**Pull request by SimeonEhrig**
Changing the OS host compiler has dangerous side effects and is not common on productive systems. Therefore our CI should not do it. Instead the CI should use  `CMAKE_CXX_COMPILER` to set the compiler.

- Instead changing the default host compiler, set the compiler via CMAKE_CXX_COMPILER argument at CMake configure.
- Rename environment variable CXX to ALPAKA_CI_CXX, because CXX can cause unexpected side effects.
- The environment variable CMAKE_CXX_COMPILER is created in the scripts depending on the variable ALPAKA_CI_CXX.
- Remove the environment variable CC completely because we do not compile C code.
- Set also CMAKE_CUDA_COMPILER as CMake configure argument manually.

Set `CMAKE_CUDA_COMPILER` as CMake argument was also necessary because of the changed behavior of setting the C++ compiler.

### CODE REVIEW

File: script/docker_ci.sh [hunk 1]
```diff
@@ -12,8 +12,8 @@ ALPAKA_CI_BOOST_BRANCH="boost-${ALPAKA_BOOST_VERSION}"
 
 # runtime and compile time options
 ALPAKA_DOCKER_ENV_LIST=()
-ALPAKA_DOCKER_ENV_LIST+=("--env" "CC=${CC}")
-ALPAKA_DOCKER_ENV_LIST+=("--env" "CXX=${CXX}")
+ALPAKA_DOCKER_ENV_LIST+=("--env" "ALPAKA_CI_CC=${ALPAKA_CI_CC}")
```
Comment (psychocoderHPC) [inline script/docker_ci.sh]: setting the C compiler could be skipped we do not use the C compiler

File: script/gitlabci/job_base.yml [hunk 2]
```diff
@@ -18,8 +18,8 @@
   image: ubuntu:${ALPAKA_CI_UBUNTU_VER}
   variables:
     ALPAKA_CI_UBUNTU_VER: "20.04"
-    CC: gcc
-    CXX: g++
+    ALPAKA_CI_CC: gcc
```
Comment (psychocoderHPC) [inline script/gitlabci/job_base.yml]: setting the C compiler should not be required an will automaticlally fall back to the system C compiler, alpaka is not compiling C code.

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

File: script/docker_ci.sh (hunk 1 above)
Comment (SimeonEhrig) [reply script/docker_ci.sh]: The boost install script is the only one, which use the variable `ALPAKA_CI_CC`. I can check, if we can use the `ALPAKA_CI_CXX` instead.

Comment (SimeonEhrig) [reply script/docker_ci.sh]: I use the `ALPAKA_CI_CXX` variable to select the C compiler in the `install_boost.sh` script. Therefore I removed the variable `ALPAKA_CI_CC` and also `CC`.

**Review by psychocoderHPC — APPROVED** [review]

**Comment by SimeonEhrig** [comment]
@psychocoderHPC ready to merge

**Pull request merged**